tests/checks/mock/test_fail2ban.py

```diff
@@ -0,0 +1,167 @@
+# stdlib
+import unittest
+import subprocess
+
+# third party
+import mock
+
+# project
+from tests.checks.common import AgentCheckTest, get_check
+
+
+class Fail2BanTestCase(unittest.TestCase):
+    def setUp(self):
+        self.config = """
+init_config:
+
+instances:
+   - sudo: True
+"""
+
+    def _get_check(self):
+        check, _ = get_check("fail2ban", self.config)
```

Comment:
Additionally subclassing `AgentCheckTest` will make the instantiated check available as `self.check` so you might not need this function after all